pluma/pluma-utils.c

```diff
@@ -79,19 +79,22 @@ pluma_utils_uri_has_file_scheme (const gchar *uri)
 gboolean
 pluma_utils_uri_has_writable_scheme (const gchar *uri)
 {
+	/* FIXME: Avoid making a new settings variable here. */
```

Comment:
I couldn't test it because this feature is no longer used.
```patch
diff --git a/pluma/pluma-settings.c b/pluma/pluma-settings.c
index eaf373e..0d1334c 100644
--- a/pluma/pluma-settings.c
+++ b/pluma/pluma-settings.c
@@ -1105,13 +1105,14 @@ pluma_settings_set_list (GSettings *settings, const gchar *key, GSList *list)
 }
 
 GSList *
-pluma_settings_get_writable_vfs_schemes (GSettings *settings)
+pluma_settings_get_writable_vfs_schemes (PlumaSettings *self)
 {
        GSList *strings;
 
        pluma_debug (DEBUG_PREFS);
 
-       strings = pluma_settings_get_list (settings, PLUMA_SETTINGS_WRITABLE_VFS_SCHEMES);
+       strings = pluma_settings_get_list (self->priv->editor_settings,
+                                          PLUMA_SETTINGS_WRITABLE_VFS_SCHEMES);
 
        /* The 'file' scheme is writable by default. */
        strings = g_slist_prepend (strings, g_strdup ("file"));
diff --git a/pluma/pluma-settings.h b/pluma/pluma-settings.h
index 4601b76..5cec0cc 100644
--- a/pluma/pluma-settings.h
+++ b/pluma/pluma-settings.h
@@ -91,7 +91,7 @@ void                   pluma_settings_set_wrap_mode                   (GSettings    *settings,
                                                                         const gchar  *key,
                                                                         GtkWrapMode   mode);
 
-GSList *               pluma_settings_get_writable_vfs_schemes         (GSettings *settings);
+GSList *                 pluma_settings_get_writable_vfs_schemes        (PlumaSettings *self);
 
 #define PLUMA_SCHEMA_ID                                        "org.mate.pluma"
 
diff --git a/pluma/pluma-utils.c b/pluma/pluma-utils.c
index 445f15a..3fc4890 100644
--- a/pluma/pluma-utils.c
+++ b/pluma/pluma-utils.c
@@ -80,13 +80,13 @@ gboolean
 pluma_utils_uri_has_writable_scheme (const gchar *uri)
 {
        /* FIXME: Avoid making a new settings variable here. */
-       GSettings *settings;
+       PlumaSettings *settings;
        GFile *gfile;
        gchar *scheme;
        GSList *writable_schemes;
        gboolean res;
 
-       settings = g_settings_new (PLUMA_SCHEMA_ID);
+       settings = _pluma_settings_get_singleton ();
        gfile = g_file_new_for_uri (uri);
        scheme = g_file_get_uri_scheme (gfile);
 
@@ -104,7 +104,6 @@ pluma_utils_uri_has_writable_scheme (const gchar *uri)
        g_slist_free_full (writable_schemes, g_free);
 
        g_free (scheme);
-       g_object_unref (settings);
 
        return res;
 }
```